autograd/functional.py

```diff
@@ -29,7 +29,7 @@ def sigmoid(x: Tensor) -> Tensor:
     Sigmoid activation function
     """
     # 709 is the maximum value that can be passed to np.exp without overflowing
-    out = Tensor(1 / (1 + np.exp(np.clip(-x.data, -709, 709))), prev=(x,))
+    out = Tensor(1 / (1 + np.exp(np.clip(-x.data, -709, 709))), prev={x})
```

Comment:
Fixing this `prev` pointer, that was previously defined as a tuple, but it should be a `set`